lighthouse-core/lib/traces/tracing-processor.js

```diff
@@ -205,23 +205,12 @@ class TraceProcessor {
    * @return {{durations: !Array<number>, clippedLength: number}}
    */
   static getMainThreadTopLevelEventDurations(model, trace, startTime, endTime) {
-    // Find the main thread via the first TracingStartedInPage event in the trace
-    const startEvent = trace.traceEvents.find(event => {
-      return event.name === 'TracingStartedInPage';
-    });
-    const mainThread = TraceProcessor._findMainThreadFromIds(model, startEvent.pid, startEvent.tid);
+    const slices = TraceProcessor.getMainThreadTopLevelEvents(model, trace, startTime, endTime);
 
     // Find durations of all slices in range of interest.
-    // TODO(bckenny): filter for top level slices ourselves?
     const durations = [];
     let clippedLength = 0;
-    mainThread.sliceGroup.topLevelSlices.forEach(slice => {
-      // Discard slices outside range.
-
-      if (slice.end <= startTime || slice.start >= endTime) {
-        return;
-      }
-
+    slices.forEach(slice => {
```

Comment:
oh puh-lease if we cared about perf here it wouldn't have taken months to realize the trace model was being constructed *multiple* times let alone that TTI runs this forEach, filter, and sort about 100 times more than necessary :P